crates/miden-lib/asm/miden/note.masm

```diff
@@ -336,3 +339,36 @@ export.add_note_assets_to_account
     # clear the stack
     drop dropw drop
 end
+
+#! Writes the assets data stored in the advice map to the memory by the provided destination
+#! pointer. 
+#!
+#! Inputs:
+#!   Operand stack: [ASSETS_COMMITMENT, num_assets, note_index, dest_ptr]
```

Comment:
We don't really need the `note_index` in this procedure, right? If so, could we refactor it so that we don't need to assume that there is no extra parameter on the stack? I'm assuming this is a matter of a couple of extra cycles - but let me know if not.